### PR TITLE
New ADT structure

### PR DESCRIPTION
**Pull request by keremispirli**
PR currently for easy review and CodeClimate.

### CODE REVIEW

File: sys/front/src/TxsHappy.y [hunk 1]
```diff
@@ -362,73 +337,73 @@ TorXakisDefn    -- :: { ( [ADTDef Name], [(Ident, TxsDef)] ) }
...
                 {  $1.inhNodeUid   = $$.inhNodeUid + 1
                 ;  $$.synMaxUid    = $1.synMaxUid
                 ;  $$.synSigs      = $1.synSigs
                 ;  $1.inhSigs      = $$.inhSigs
-                ;  $$ = [ $1 ]
+                ;  $$ = mkLegacyDefinitions [ $1 ]
                 }
 
-TypeDefList     -- :: { [ (Ident,TxsDef) ] }
+TypeDefList     -- :: { [ ADTDef Name ] }
```
Comment (keremispirli) [inline sys/front/src/TxsHappy.y]: It should also be `ADTDef (Either Sort Name)`, like LN418

**Comment by dnadales** [comment]
Can we close this pull request for now?

**Comment by dnadales** [comment]
> Can we close this pull request for now?

Pretty please?

**Comment by keremispirli** [comment]
Closing due to pretty please :) Will reopen when we can work on this (i.e. when new compiler is ready).

**Pull request closed without merging**